### DISCUSSION TITLE
Can the html-audio-response audio be exported as a separate file?

### DISCUSSION THREAD

**Discussion by wj-mitchell** [jspsych/jsPsych#3393] [Q&A]
Hi all,

I would like to use a custom version of the html-audio-response plugin to record free recall data which could last up to 20 minutes. This seems to pose an issue for the integrity of all of the other data when output as a single .csv, given the base 64 string formatted audio data and the limitations of how many characters can be contained in a single cell. I know if you `allow_playback: true`, you have an option to download the data as a separate .weba file. I have dumped a lot of time in trying to modify the plugin to export this automatically with no success. However, I know jsPsych.data allows for exporting .json files in addition to .csv files, which seemingly might be a safe way to store these base 64 strings from what I've read and might keep my .csv data in tact if the audio data was stored there. 

Is it possible to export all of my other experimental data as a .csv file, my audio data as a stand alone .json file, and still be able to convert that .json file to an audio recording (e.g., .mp3 format) later with Python or R? As always, thanks for your time!

**Comment by ChristopheBossens** [comment]
Hi wj-mitchell,

Have you tried the example on the html-audio-response page where you save the audio file to the server after the trial and only log an id (which you can add to your csv)? 

I have done this conversion in the past (saving the base64 string and converting back to an audio file) so it is possible. Let me know if you are able to save the audio file first, I can then look up the code again to recover the audio file :)
